src/ethereum/ethEngine.js

```diff
@@ -322,6 +322,31 @@ export class EthereumEngine extends CurrencyEngine {
   async makeSpend(edgeSpendInfoIn: EdgeSpendInfo) {
     const { edgeSpendInfo, currencyCode } = super.makeSpend(edgeSpendInfoIn)
 
+    /**
+    For RBF transactions, get the gas price and limit (fees) of the existing 
+    transaction as well as the current nonce. The fees and the nonce will be 
+    used instead of the calculated equivalents.
+    */
+    let rbfGasPrice: string
+    let rbfGasLimit: string
+    let rbfNonce: string
+    const rbfTxid =
+      edgeSpendInfo.rbfTxid && normalizeAddress(edgeSpendInfo.rbfTxid)
+    if (rbfTxid) {
+      const rbfTxIndex = this.findTransaction(currencyCode, rbfTxid)
+
+      if (rbfTxIndex > -1) {
+        const rbfTrx = this.transactionList[currencyCode][rbfTxIndex]
+
+        if (rbfTrx.otherParams) {
+          const { gasPrice, gas, nonceUsed } = rbfTrx.otherParams
+          rbfGasPrice = bns.mul(gasPrice, '2')
+          rbfGasLimit = gas
+          rbfNonce = nonceUsed
+        }
+      }
```

Comment:
Need to fail the make spend if the RBFTX ID cannot be found.

Reply:
As well if the other parameters cannot be found

Reply:
Otherwise we will end up sending a duplicate transaction that the user did not intend

Reply:
Rebased the commit to include check that throws an exception when data is missing. This catches both the case where the transaction is not found or the data on a found transaction is missing.

https://github.com/EdgeApp/edge-currency-accountbased/pull/213/commits/bb5e3b7f48bda4de9ee9a9f885e6a372bcfe3d41#diff-4839c75ac65fd8060df92dcdc5cb0ab5786e8fa09c3ba01287bc97a2c53cf971R349-R351